app/controllers/transactions_controller.rb

```diff
@@ -2,7 +2,13 @@ class TransactionsController < ApplicationController
   before_action :find_transaction, only: %i[update edit]
 
   def index
-    @transactions = current_user.transactions.paginate(page: params[:page], per_page: 10)
+    @search = current_user.transactions.ransack(params[:q])
+    @transactions = @search.result(distinct: true).paginate(page: params[:page], per_page: 10)
+  end
+
+  def search
+    index
```

Comment:
@marderer as I understand you need `@search` and `@transactions`  in `#index` as well as in `#serch`

For that you can add before action:
```ruby
before_action :search_for_transactions, only: %i[index search]

def index
end

def search
  render :index
end

private

def search_for_transactions
  @search = current_user.transactions.ransack(params[:q])
  @transactions = @search.result(distinct: true).paginate(page: params[:page], per_page: 10)
end
```

Reply:
@marderer before making those changes be sure to add controller test to your `spec/controllers` folder. All actions from controller have to be tested with different types of data.

Reply:
@AleksSenkou I added some tests for activity page. I think for CRUD actions better add where I adding this actions to activity page. ok?

Reply:
@marderer sorry, I can't get what's your question